packages/afl-persistent/afl-persistent.1.4/opam

```diff
@@ -10,7 +10,7 @@ build: [
   [ "./config.sh" ]
 ]
 patches: [
-  "config.patch"
+  ["config.patch" "md5=ff93f594181d6508acdce88597e3c7b6"]
```

Comment:
```suggestion
  "config.patch"
]  
extra-files: [
   ["config.patch" "md5=ff93f594181d6508acdce88597e3c7b6"]
```

Reply:
including the checksum directly in the `patches` section is not valid opam syntax

Reply:
Aaah, thanks, that is why I had not found it in the docs when I made the PR. So we will need to add all patches both as patches and as extra files? Wouldn't make sense to add the hash also for the patches?

Reply:
yes, well -- there are various paths forward:
- the whole extra-files stuff is quite cumbersome in various areas of opam, maybe it can be sunset (while retaining the option to add patches as extra-sources)?
- another option is to include these patch files directly in the opam file (they'd grow in size)
- in early 2022, Reynir and myself spent a day to figure out what the semantics / intentions of opam and checksums (and extra-files) is, and we discovered some inconsistent behaviour - it may be worth to put some re-thinking on a future roadmap (2.3? 3.0?)
- the checksums were added for repository signing, but as pointed in the refered PR, the data in `files` is copied independent of whether it is listed in the opam file --> this means for repository signing we can't rely on that fact (or need to wait until the PR mentioned is merged, and opam will be more strict).

but, with the current state of affairs, it's best to put the patches and extra-files in the opam file :)

Reply:
Makes sense, thanks for the context :)